packages/@uppy/companion/src/server/provider/drive/index.js

```diff
@@ -142,8 +147,25 @@ class Drive extends Provider {
           .request()
       }
 
+      let stopDataTransfer = false
       requestStream
-        .on('data', (chunk) => onData(null, chunk))
+        .on('response', (resp) => {
+          if (resp.statusCode !== 200) {
+            stopDataTransfer = true
```

Comment:
Nit/A possible simplification: perhaps we could add the `.on('data')` handler inside this function, only if the response status was 200, instead of keeping this separate variable?
```js
requestStream.on('response', (resp) => {
  if (resp.statusCode === 200) {
    requestStream.on('data', chunk => onData(null, chunk))
  } else {
    // handle error
  }
})
```

Reply:
went ahead with this suggestion, thank you! 👍